internal/db/job_tracker.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2025 Canonical.
+
+package db
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/servermon"
+)
+
+// AddJob adds a new job to the job tracker.
+// It returns the job ID and an error if the job already exists or if the creation fails
+func (d *Database) AddJob(ctx context.Context, jobType string) (jobId uuid.UUID, err error) {
+	const op = errors.Op("db.AddJob")
+	if err := d.ready(); err != nil {
+		return jobId, errors.E(op, err)
+	}
+
+	durationObserver := servermon.DurationObserver(servermon.DBQueryDurationHistogram, string(op))
+	defer durationObserver()
+	defer servermon.ErrorCounter(servermon.DBQueryErrorCount, &err, string(op))
+
+	db := d.DB.WithContext(ctx)
+
+	entry := dbmodel.NewJobTrackerEntry(jobType)
+	if err := db.Create(entry).Error; err != nil {
+		err := dbError(err)
+		if errors.ErrorCode(err) == errors.CodeAlreadyExists {
```

Comment:
afaics this would never return this type of error as we create the entry on L32 - it will be assigned a unique UUID (unlikely to clash with an existing one).

Reply:
Yup, basically impossible but good to catch it. Be hilarious if we ever saw this.